### DISCUSSION TITLE
[GSoC 2026] Alaska Wildfire Prediction MVP Complete - Seeking Feedback on Multi-Modal Fusion

### DISCUSSION THREAD

**Discussion by farhann-saleem** [YaliWang2019/AK-Satellite-Imagery-Wildfire-Prediction#27] [Show and tell]
Hi @YaliWang2019 ,

**Farhan Saleem** here, a CS student (3rd semester) with a background in Data Engineering, Machine Learning, backend, and AWS. I've successfully completed **Phase 1 (MVP)** of the Alaska Wildfire Prediction project.

### 🚀 Phase 1 Status: MVP & ETL Pipeline
I built a fully functioning **ETL pipeline** that automates the ingestion of Sentinel-2 imagery and MTBS burn severity maps via Google Earth Engine. The system now successfully:
* Exports aligned, cloud-masked imagery
* Extracts **64×64** training patches
* Normalizes and prepares tensors for the model

###  Key Technical Challenge: The "Lazy Logic" Trap
During initial baseline modeling, I encountered a critical failure:
* **The Problem:** Due to severe class imbalance (**98.3% No Fire** vs **1.7% Fire**), the baseline CNN learned "lazy logic" .predicting "No Fire" for every pixel to maximize accuracy (98%) while achieving **0.0% Recall**.
* **The Solution:** I implemented sample weighting (10× boost for fire class) to force the model to focus on hard-to-classify fire pixels.
* **Result:** This intervention boosted Recall from **0% to 58.6%** 

###  Repository & Development Approach
I developed this MVP in my personal repository to rigorously stress-test different architectures and identify potential flaws (like the gradient explosion I faced earlier) before proposing a final solution.

Now that the core architecture is validated, I'm opening it for review:
**Project Repository:** https://github.com/farhann-saleem/Alaska-Wildfire-prediction-MVP

 **Full Documentation:** https://github.com/farhann-saleem/Alaska-Wildfire-prediction-MVP/tree/main/docs

I've attached my handwritten architectural notes and problem analysis below:
### Training Results
<img width="800" alt="training_results" src="https://github.com/user-attachments/assets/36937c28-395a-4df0-bfc8-3af68cd1688b" />
### Handwritten Architecture Notes
<table>
  <tr>
    <td><img src="https://github.com/user-attachments/assets/e372e92a-afaa-4a9d-a444-b9598b19a4eb" alt="Phase 1 Page 1" width="400"/></td>
    <td><img src="https://github.com/user-attachments/assets/fee421a9-f70c-4d7b-a134-eb0cc1048b58" alt="Phase 1 Page 2" width="400"/></td>
  </tr>
  <tr>
    <td align="center"><b>Architecture & Pipeline</b></td>
    <td align="center"><b>Problem & Solution</b></td>
  </tr>
</table>

###  Next: Phase 2
I'm planning to experiment with **CNN-LSTM** for temporal analysis and **multi-modal fusion** (Sentinel-1 SAR + weather data).

**Questions for mentors:**
1. Would Sentinel-1 integration be valuable for Alaska's cloudy conditions?
2. CNN-LSTM vs Transformers for temporal patterns.Any recommendations?

Looking forward to your feedback!

**Comment by YaliWang2019** [comment]
Hi Farhan @farhann-saleem,

Thanks for sharing this MVP and the detailed write-up, this is impressive and solid work. It’s clear you didn’t just assemble a pipeline, but actually spent time understanding why the model failed initially and how to fix it. Your analysis of the class imbalance issue and the “lazy model” behavior shows good intuition about optimization dynamics and loss behavior. I like the way you documented the failure modes (softmax collapse, gradient instability) and the reasoning behind your fixes.

As we move beyond Phase 1, I’d like to deliberately shift the focus from “can we make a model work?” to “what are we actually learning about wildfire drivers?” In other words, Phase 2 should be less about stacking additional modalities and more about forming and testing research hypotheses.

Proposed Phase 2 (research-oriented task):
Instead of immediately integrating Sentinel-1 or moving to CNN-LSTM, I’d like you to start with the weather side of the problem. Please identify ~5 ERA5 variables that you believe are genuinely relevant to wildfire ignition and/or spread in Alaska, and briefly explain for each: 1). why it matters physically or ecologically, 2). whether it is likely a direct contributing factor, a correlated proxy, or part of a causal chain, and 3). at what temporal scale it is most meaningful (e.g., short-term extremes vs. longer-term conditions).

You do not need to run experiments at this stage. A concise, well-reasoned write-up (with references if helpful) is sufficient; it doesn't need to be very long. The goal here is to make sure that when we introduce multi-modal fusion later, it’s driven by scientific reasoning rather than architectural complexity.

Once you’ve done this, we can discuss how and whether these variables should be integrated with the Sentinel-based pipeline you’ve already built, and what kind of temporal modeling actually makes sense for the questions we care about. I’m looking forward to seeing how you approach the problem from a more hypothesis-driven perspective in Phase 2. Please feel free to let me know your thoughts or questions.

Thanks and best,
Yali Wang

**Comment by farhann-saleem** [comment]
Subject: Re: Phase 2 Guidance - Moving to Hypothesis-Driven Analysis

Hi @YaliWang2019 , 

Thank you for the detailed feedback. I completely agree with the shift in focus moving from "stacking modalities" to a "hypothesis-driven" understanding of the physical drivers is the right scientific approach.

I want to ensure my analysis of the ERA5 variables is backed by evidence, not just theory. Instead of simply listing variables, I am currently running a script to probe the ERA5 historical data for the specific fire patches I identified in Phase 1.

I plan to generate correlation plots for variables like volumetric_soil_water and temperature_2m against the actual ignition dates from my MVP dataset. This will allow me to demonstrate empirically which factors were the strongest drivers for these specific Alaskan fires.

I will compile this analysis and share the write-up early next week.

Best, Farhan

**Comment by farhann-saleem** [comment]
# Phase 2 Analysis: The "Low VPD" Paradox & Energy-Efficiency Strategy

**Author:** Farhan Saleem  
**Status:**  Phase 2 Complete (Empirical Verification)

---

##  Executive Summary: Aligning with Wang et al. (2023)

Hi  DR .@YaliWang2019  , In *Toward Energy-Efficient Deep Neural Networks for Forest Fire Detection* (2023), you highlighted the critical bottleneck of **on-board energy consumption** for drone-based detection.

**My Phase 2 analysis of 511 Alaska fires confirms a novel way to solve this:**
We found that Alaska fires consistently ignite at an unusually low **Vapor Pressure Deficit (VPD)** threshold of **0.70 kPa** (vs. >1.5 kPa in temperate zones).

**Strategic Proposal:** This finding enables a **"Low-Power Wake-Up" architecture**. Instead of running heavy CNNs continuously, we can use this specific VPD signature as a computationally "free" gate, activating the GPU only when risks exist. This directly addresses the energy-latency trade-off discussed in your work.

---

##  Empirical Findings (N=511)

To validate this, I re-architected our pipeline (ERA5-Land + NASA GPM Hybrid) to rigorously test pre-fire conditions.
Code: [scripts/era5_analysis.py](https://github.com/farhann-saleem/Alaska-Wildfire-prediction-MVP/blob/main/scripts/era5_analysis.py)
Full Repo: [Alaska-Wildfire-prediction-MVP](https://github.com/farhann-saleem/Alaska-Wildfire-prediction-MVP)
Data: [results/phase2/era5_weather_correlations.csv](https://github.com/farhann-saleem/Alaska-Wildfire-prediction-MVP/blob/main/results/phase2/era5_weather_correlations.csv)

| Metric | Alaska Mean | Standard Fire Model | Diagnosis |
| :--- | :--- | :--- | :--- |
| **VPD** | **0.70 kPa** | > 1.5 kPa | 📉 **Anomaly:** Fires burn in "safe" air. |
| **Precipitation** | **30.9 mm** | < 10 mm | ☁️ **Anomaly:** Fires burn in "wet" months. |
| **Temperature** | **15.7°C** | > 25°C | 🌡️ **Anomaly:** Fires burn in cool weather. |

> **Scientific Conclusion:** Alaska wildfires are mechanistically distinct. They are driven by **long-term fuel drying** (moss/lichen dynamics), not immediate atmospheric heat. Standard "Hot/Dry" detection models will fail here.

---

## Evidence Visualizations
*(N=511 Validated Burn Patches)*

<table>
  <tr>
    <td align="center" width="50%">
      <img src="https://github.com/user-attachments/assets/531b8804-6e2f-44fe-8ba4-7c03df9337c3" alt="Correlation Heatmap" width="100%">
      <br><b>VPD-Temperature Coupling</b>
    </td>
    <td align="center" width="50%">
      <img src="https://github.com/user-attachments/assets/79a00f39-e93f-4928-9b6d-6e6e8a568aa6" alt="Variable Distributions" width="100%">
      <br><b>The "Cool & Wet" Distribution</b>
    </td>
  </tr>
  <tr>
    <td colspan="2" align="center">
      <img src="https://github.com/user-attachments/assets/b12f950e-e887-45f1-ae3d-37e74347f228" alt="VPD Distribution" width="60%">
      <br><b>3. The "Alaska Anomaly" (VPD < 0.70 kPa)</b>
      <br><i>(This specific plot proves the "Low-Power Wake-Up" threshold viability)</i>
    </td>
  </tr>
</table>



##  Phase 3 Proposal: Validation Before Architecture

### Research-First Approach

Before proceeding with multi-modal fusion, I propose validating whether the "Low VPD Paradox" holds in **out-of-sample data (2022-2023)**. This ensures our Alaska-specific patterns are reproducible and not artifacts of the 2015-2020 training period.

### Proposed Phase 3 Strategy:


#### **Step 1: Temporal Validation**
*   Extract MTBS fire events from 2022-2023.
*   Probe ERA5 at these locations using identical protocol.
*   Test if VPD threshold (~0.70 kPa) remains stable.
*   **Deliverable:** Statistical validation report.
#### **Step 2: Architecture Design (Conditional)**
*   **If validation confirms pattern stability:**
    *   Proceed with energy-efficient "wake-up" architecture.
    *   Integrate Sentinel-1 SAR for all-weather capability.
    *   Evaluate Contextual Transformer backbone (Ding et al., 2025).
*   **If patterns diverge:**
    *   Investigate why (climate shift vs. sampling bias).
    *   Adjust research direction accordingly.

---

##  Key Question for Guidance

The 0.70 kPa VPD threshold is counterintuitive compared to temperate fire models. Before building architecture around it:
**Should I prioritize temporal validation (2022-2023 data) to confirm this pattern is stable, or proceed directly to multi-modal fusion assuming it holds?**
I recognize this is fundamentally a research question.I want to ensure the finding is reproducible before committing to complex architectures.

---

Thank you for the research-oriented direction in Phase 2. Looking forward to your guidance on the validation approach.
I am actively exploring research papers to yield the best output possible. I am currently evaluating the **Contextual Transformer** block (Ding et al., 2025) (*"Multi-Scale Enhanced Contextual Transformer Network for forest fire detection"*) as a potential backbone replacement to better capture the subtle features of these "low-intensity" ignition zones.
**Best regards,**
Farhan Saleem

**Comment by YaliWang2019** [comment]
Hi Farhan,

Thank you so much for the detailed response to the task. It's pretty impressing. Once the proposal submission is open, we'd love to see your submitted proposal if you're still interested in working on this project.

Thanks and best,
Yali Wang
